### PR TITLE
Refactor additional information

### PR DESCRIPTION
**Pull request by serucee**

### CODE REVIEW

File: wirecard-woocommerce-extension/classes/helper/class-additional-information.php [hunk 1]
```diff
@@ -320,55 +425,39 @@ public function map_state_to_iso_code( $country, $state ) {
...
 	 */
-	public function create_basket_from_parent_transaction( $order, $config, $payment_method, $refund_basket = [], $refunding_amount = 0 ) {
+	public function create_basket_from_parent_transaction( $order, $config, $payment_method, $refund_basket = array(), $refunding_amount = 0 ) {
 		$basket              = new Basket();
 		$transaction_service = new \Wirecard\PaymentSdk\TransactionService( $config );
 		$parent_transaction  = $transaction_service->getTransactionByTransactionId( $order->get_transaction_id(), $payment_method );
 		$items_total         = 0;
 		$shipping            = 0;
 
+		// TODO: only call build_basket_item_from_array once, always use $item, remove calc logic from method call
```
Comment (flavius) [inline wirecard-woocommerce-extension/classes/helper/class-additional-information.php]: Create a ticket for this TODO and add it to the TODO or resolve it.

File: wirecard-woocommerce-extension/classes/helper/class-additional-information.php [hunk 2]
```diff
@@ -387,9 +476,9 @@ private function get_correct_country_for_tax_rate() {
 		$tax_setting = get_option( 'woocommerce_tax_based_on' );
 
 		switch ( $tax_setting ) {
-			case 'billing':
+			case self::BILLING:
 				return WC()->customer->get_billing_country();
-			case 'shipping':
+			case self::SHIPPING:
 				return WC()->customer->get_shipping_country();
 			case 'base':
```
Comment (flavius) [inline wirecard-woocommerce-extension/classes/helper/class-additional-information.php]: Introduce symbolic constant.

File: wirecard-woocommerce-extension/classes/includes/class-wc-gateway-wirecard-giropay.php [hunk 3]
```diff
@@ -88,81 +88,81 @@ public function __construct() {
 	 * @since 2.2.0
 	 */
 	public function init_form_fields() {
-		$this->form_fields = [
-			'enabled'             => [
+		$this->form_fields = array(
```
Comment (flavius) [inline wirecard-woocommerce-extension/classes/includes/class-wc-gateway-wirecard-giropay.php]: Why did we change from `[` to `array(`? There's probably no right or wrong, just curious about the reasoning.

File: wirecard-woocommerce-extension/classes/helper/class-additional-information.php [hunk 4]
```diff
@@ -232,6 +233,110 @@ public function create_address_data( $order, $type ) {
...
+	 * @param float $tax_rate
+	 * @param null|float $tax_amount
+	 * @param null|string $currency
+	 *
+	 * @return Item
+	 *
+	 * @since 3.1.0
+	 */
+	protected function build_basket_item( $name, $amount, $quantity, $description, $article_number, $tax_rate, $tax_amount = null, $currency = null ) {
+		// TODO: move basket logic into separate class
```
Comment (flavius) [inline wirecard-woocommerce-extension/classes/helper/class-additional-information.php]: Write a ticket for this and reference it here.

File: wirecard-woocommerce-extension/classes/includes/class-wc-gateway-wirecard-giropay.php (hunk 3 above)
Comment (serucee) [reply wirecard-woocommerce-extension/classes/includes/class-wc-gateway-wirecard-giropay.php]: phpcs did it automatically, so i assume there was an update in wordpress CS for this

**Pull request closed without merging**